### DISCUSSION TITLE
Tailwind CSS v4 will probably not support Alpine (musl libc)

### DISCUSSION THREAD

**Discussion by flavorjones** [flavorjones/tailwindcss-ruby#22] [General]
👋 Maintainer here.

I started preparing for the [imminent beta release of Tailwind CSS 4](https://x.com/adamwathan/status/1857921557747650572), and ended up [opening an upstream bug report to raise awareness that the alphas do not support Alpine/Musl](https://github.com/tailwindlabs/tailwindcss/issues/15031).

Given [this response from the bun maintainers](https://github.com/oven-sh/bun/issues/14292#issuecomment-2388252720), it seems unlikely that Tailwind 4 will run on Alpine anytime soon.

I'm just posting this to let people know what to expect. I'll try to keep the thread updated as I learn more.

**Comment by flavorjones** [comment]
Upstream progress:
- https://github.com/tailwindlabs/tailwindcss/pull/15567

The next upstream v4 release should have musl-compatible binaries!

**Comment by flavorjones** [comment]
https://github.com/flavorjones/tailwindcss-ruby/releases/tag/v4.0.0.beta.9 is out with musl support!

**Comment by excid3** [comment]
v4 was just released. https://github.com/tailwindlabs/tailwindcss/releases/tag/v4.0.0

**Reply by flavorjones** [reply]
https://github.com/flavorjones/tailwindcss-ruby/pull/46

**Reply by flavorjones** [reply]
https://github.com/flavorjones/tailwindcss-ruby/releases/tag/v4.0.0
